frontend/web/components/EnvironmentVersioningListener.tsx

```diff
@@ -0,0 +1,34 @@
+import { FC, useEffect } from 'react'
+import { useGetEnvironmentQuery } from 'common/services/useEnvironment'
+import AppActions from 'common/dispatcher/app-actions'
+type EnvironmentVersioningListenerType = {
+  id: number
+  versioningEnabled: boolean
+  onChange: () => void
+}
+
+const EnvironmentVersioningListener: FC<EnvironmentVersioningListenerType> = ({
+  id,
+  onChange,
+  versioningEnabled,
+}) => {
+  const { data, isFetching, isSuccess } = useGetEnvironmentQuery(
+    { id: `${id}` },
+    {
+      pollingInterval: versioningEnabled ? undefined : 2000,
```

Comment:
This ternary feels like it's the wrong way around to me 🤔 ? Aren't we trying to enable polling if versioning is enabled? If so, shouldn't we define the polling interval in this case?